core/src/blocktree.rs

```diff
@@ -2065,6 +2127,63 @@ pub mod tests {
         Blocktree::destroy(&blocktree_path).expect("Expected successful database destruction");
     }
 
+    #[test]
+    fn test_detached_head() {
+        let blocktree_path = get_tmp_ledger_path("test_is_detached_head");
+        {
+            let blocktree = Blocktree::open(&blocktree_path).unwrap();
+
+            // Create blobs and entries
+            let entries_per_slot = 1;
+            let (blobs, _) = make_many_slot_entries(0, 3, entries_per_slot);
+
+            // Write slot 2, which chains to slot 1. We're missing slot 0,
+            // so slot 1 is the detached head
+            blocktree.write_blobs(once(&blobs[2])).unwrap();
+            let meta = blocktree
+                .meta(1)
+                .expect("Expect database get to succeed")
+                .unwrap();
+            assert!(Blocktree::is_detached_head(&meta));
+            assert_eq!(get_detached_heads(&blocktree), vec![1]);
+
+            // Write slot 1 which chains to slot 0, so now slot 0 is the
+            // detached head, and slot 1 is no longer the detached head.
+            blocktree.write_blobs(once(&blobs[1])).unwrap();
+            let meta = blocktree
+                .meta(1)
+                .expect("Expect database get to succeed")
+                .unwrap();
+            assert!(!Blocktree::is_detached_head(&meta));
+            let meta = blocktree
+                .meta(0)
+                .expect("Expect database get to succeed")
+                .unwrap();
+            assert!(Blocktree::is_detached_head(&meta));
+            assert_eq!(get_detached_heads(&blocktree), vec![0]);
+
+            // Write some slot that also chains to existing slots and detached head,
+            // nothing should change
+            let blob4 = &make_slot_entries(4, 0, 1).0[0];
+            let blob5 = &make_slot_entries(5, 1, 1).0[0];
+            blocktree.write_blobs(vec![blob4, blob5]).unwrap();
+            assert_eq!(get_detached_heads(&blocktree), vec![0]);
+
+            // Write zeroth slot, no more detached heads
+            blocktree.write_blobs(once(&blobs[0])).unwrap();
+            for i in 0..3 {
+                let meta = blocktree
+                    .meta(i)
+                    .expect("Expect database get to succeed")
+                    .unwrap();
+                assert!(!Blocktree::is_detached_head(&meta));
+            }
+            // Detached heads is empty
```

Comment:
clear zero, what happens?

Reply:
As in deleting slot zero? The detached heads remains unchanged, no cascading deletes